pkg/provisioner/ironic/adopt_test.go

```diff
@@ -73,6 +73,7 @@ func TestAdopt(t *testing.T) {
 
 			expectedDirty:        false,
 			expectedRequestAfter: 0,
+			expectedError:        true,
 		},
 	}
 
```

Comment:
Do we still need a test case for retry and to make sure that Node will move from AdoptFail state back to manageable?